JASP-Engine/JASP/R/networkanalysis.R

```diff
@@ -0,0 +1,2132 @@
+		# general arguments
+		"weightedNetwork", "signedNetwork", "missingValues"
+	)
+	bootstrapArgs <-c(defArgs, "numberOfBootstraps", "BootstrapType")
+
+	stateKey <- list(
+		# depends on everything but plotting arguments
+		network = defArgs,
+		# depends also on bootstrap options
+		bootstrap = bootstrapArgs,
+		# depends also on normalization of centrality measures
+		centrality = c(defArgs, "normalizeCentrality", "maxEdgeStrength", "minEdgeStrength"),
+		# depends also on plotting arguments
+		networkPLT = c(defArgs,
+					   # depends also on plotting arguments
+					   "plotWidthNetwork", "plotHeightNetwork",
+					   "layout", "edgeColors", "repulsion", "edgeSize", "nodeSize", "colorNodesBy",
+					   "maxEdgeStrength", "minEdgeStrength", "cut", "showDetails", "nodeColors",
+					   "showLegend", "legendNumber", "showMgmVariableType", "showVariableNames",
+					   "graphSize", "scaleLabels", "labelSize", "abbreviateLabels", "abbreviateNoChars",
+					   "keepLayoutTheSame", "layoutX", "layoutY"),
+		centralityPLT = c(defArgs,
+						  "plotWidthCentrality", "plotHeightCentrality", "normalizeCentrality", "abbreviateLabels", "abbreviateNoChars"),
+		layout = c(defArgs, "layout", "repulsion", "layoutX", "layoutY"),
+		bootstrapEdgePLT = bootstrapArgs,
+		bootstrapCentPLT = bootstrapArgs
+	)
+
+	if (options[["keepLayoutTheSame"]]) # ensure layout remains unchanged unless the data changes
+		stateKey[["layout"]] <- "variables"
+
+	# show info about variable types. Done here so that hopefully a table gets shown even if .hasErrors finds something bad.
+	if (perform == "run" && options[["estimator"]] == "mgm" && !is.null(options[["mgmVariableTypeData"]])) {
+
+	  options[["mgmVariableTypeData"]] <- .networkAnalysisSanitizeMgmVariableType(variables = variables, options = options)
+
+		results[["generalTB"]] <- .networkAnalysisGeneralTable(NULL, dataset, options, perform = "init") # any errors will appear top of this table
+		results[["mgmTB"]] <- .networkAnalysisMgmVariableInfoTable(network, variables, options, perform)
+
+	}
+
+	## Do Analysis ## ----
+
+	# Sort out whether things are set to defaults or not.
+	if (length(variables) > 2) {
+		
+		# some analyses, such as Sacha's EBIGglasso with cor_auto, completely ignore the missing argument
+		# and always use pairwise information even though their documentation says they can do listwise
+		if (options[["missingValues"]] == "listwise")
+			dataset <- dataset[complete.cases(dataset), ]
+
+		# check for errors, but only if there was a change in the data (which implies state[["network"]] is NULL)
+		if (is.null(state[["network"]])) {
+
+		  # default error checks
+		  checks <- c("infinity", "variance", "observations")
+
+			groupingVariable <- NULL
+			if (options[["groupingVariable"]] != "") {
+				groupingVariable <- options[["groupingVariable"]]
+				# these cannot be chained unfortunately
+				.hasErrors(dataset = dataset[.v(groupingVariable)], perform = perform,
```

Comment:
Is this still needed with the more robust error handling? If so, changes must be made.

Reply:
Sure, but I'd like to test first if that that PR fixes the issues, before making the changes.